wordpress-mobile/WordPress-Android
# Changes related to deep linking email marketing

Fixes #9864

To test:
- Should not open the app: `https://wordpress.com/post/`
- Should not open the app: `https://public-api.wordpress.com/nope/`
- Should not open the app: `https://public-api.wordpress.com/bar/?redirect_to=https%3A%2F%2FCHANGEME.wordpress.com%2Fpost`
- Should open the app's editor on previously selected site: `https://wordpress.com/post/wrong-hostname`
- Should open the app's editor with selected site CHANGEME: `https://wordpress.com/post/CHANGEME.wordpress.com`
- Should open the app's editor with selected site CHANGEME: `https://public-api.wordpress.com/mbar/?redirect_to=https%3A%2F%2Fwordpress.com%2Fpost%2FCHANGEME.wordpress.com`
- Should "open" the app, but then redirect to the browser: `https://public-api.wordpress.com/mbar/?redirect_to=https%3A%2F%2Fwordpress.com%2Fnope`


<s>Note: this is a draft PR because backend isn't updated yet and the deep linking urls might changes (`mobile-api`)</s>

This can be tested with ADB:

```shell
$ adb shell am start -a android.intent.action.VIEW \
 -c android.intent.category.BROWSABLE \
 -d "https://public-api.wordpress.com/mbar/?redirect_to=https%3A%2F%2Fwordpress.com%2Fpost%2FCHANGEME.wordpress.com"
```

Real test:
- Create a new site.
- Publish a post.
- Switch to another site (to make sure the editor is opened for the right site later).
- Wait for the "Congrats for your first post" marketing email.
- Tap the "Create new post" button, you should be able to choose between your browser or the WordPress app, select the WordPress app, the editor should open on this site.

[Demo video](https://bia.is/s/lhOY/mbar-email-demo.webm) of this ^

cc @diegoreymendez
